### ISSUE TITLE
breakpad: error: field ‘context’ has incomplete type ‘google_breakpad::ucontext’      struct ucontext context;

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#56]
I got that while trying to build Daemon and Unvanquished on Ubuntu 17.10:

```
[  9%] Building CXX object CMakeFiles/srclibs-breakpad.dir/libs/breakpad/src/client/linux/crash_generation/crash_generation_server.cc.o
In file included from /home/illwieckz/dev/buildme-unvanquished/Daemon/libs/breakpad/src/client/linux/crash_generation/crash_generation_server.cc:46:0:
/home/illwieckz/dev/buildme-unvanquished/Daemon/libs/breakpad/src/client/linux/handler/exception_handler.h:194:21: error: field ‘context’ has incomplete type ‘google_breakpad::ucontext’
     struct ucontext context;
                     ^~~~~~~
/home/illwieckz/dev/buildme-unvanquished/Daemon/libs/breakpad/src/client/linux/handler/exception_handler.h:194:12: note: forward declaration of ‘struct google_breakpad::ucontext’
     struct ucontext context;
            ^~~~~~~~
```

**Comment by illwieckz** [comment]
That [random patch](https://github.com/faulesocke/void-packages/blob/master/srcpkgs/chromium/patches/0002-replace-struct-ucontext-with-ucontext_t.patch) found on _da internetz_ does not apply nicely (some changes are rejected by our current breakpad code base) but is enough to make engine and vm build. So the changes there are interesting and it's a good start.

**Comment by DolceTriade** [comment]
Oh, I hit this bug earlier. This struct was literally renamed... I haven't gotten around to patching breakpad to fix it though :(

**Comment by slipher** [comment]
According to [this](https://sourceware.org/bugzilla/show_bug.cgi?id=21457#c14) the name `ucontext_t` is also available in older versions, so you can simply s/struct ucontext/ucontext_t/ and have it work in old and new versions.

**Comment by slipher** [comment]
Fixed by b26b5b2e455ff3a4268352d7108bb69fccc6bf6b
